src/utils.js

```diff
@@ -1,3 +1,46 @@
+const NON_STRING_PLACEHOLDER = "__REACT_EMOJI_RENDER_PLACEHOLDER__";
+
 export function escapeStringToBeUsedInRegExp(string) {
   return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"); // $& means the whole matched string
 }
+
+export function stripNonStringElements(children) {
+  if (Array.isArray(children)) {
+    const stripped = children
+      .map(x => (typeof x === "string" ? x : NON_STRING_PLACEHOLDER))
+      .join("");
+    return [stripped, children.filter(x => typeof x !== "string")];
+  } else if (typeof children === "string") {
+    return [children, []];
+  }
+}
+
+export function returnStrippedElements(stripped, elements) {
+  if (!stripped || !Array.isArray(stripped)) return stripped;
+  if (!elements || !elements.length) return stripped;
+
+  let count = -1;
+  return stripped.flatMap(x => {
```

Comment:
I'm not sure if using `flatMap` is fine regarding browser support, I would probably change it with a `reduce` + `map` alternative. Do you mind ?

Also this function is a bit convoluted with up to 6 levels of nesting, can we do better for future contributors / maintainers ? 😁

Reply:
You're right, I'm used to the typescript compiler taking care of that for me. (maybe there's some babel config that can be tuned).
I'll see about splitting the function as well.